catalogue/webapp/pages/worksv2.js

```diff
@@ -34,26 +34,35 @@ export const Works = ({
   const [query, setQuery] = useState(initialQuery);
   const [works, setWorks] = useState(initialWorks);
   const [page, setPage] = useState(initialPage);
-  const pagination = works ? PaginationFactory.fromList(
-    works.results,
-    Number(works.totalResults) || 1,
-    Number(page) || 1,
-    works.pageSize || 1,
-    {query: query || ''}
-  ) : null;
-
-  // https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops
-  // Make sure if we're using the Pagination element, which reinitialised the
-  // `getInitialProps` we set the works to the `initialWorks` from that function
-  if (page !== initialPage && query === initialQuery) {
-    setPage(initialPage);
-    setWorks(initialWorks);
-  }
+  const [loading, setLoading] = useState(false);
+  // 1. We use this as we get the `initalWorks` from `getInitialProps`
+  //    Whereas standard React apps would fetch on first render
+  //    See: https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables
+  const firstRender = useRef(true);
 
   useEffect(() => {
-    // Update the document title using the browser API
     document.title = `${query} | Catalogue search | Wellcome Collection`;
-  });
+  }, [query]);
+
+  useEffect(() => {
+    // Don't run this on first render, see comment 1. above
+    if (firstRender.current) {
+      firstRender.current = false;
+      return;
+    }
+
+    setLoading(true);
+    Router.push(
+      worksV2Link({ query, page }).href,
+      worksV2Link({ query, page }).as,
+      { shallow: true }
+    );
+    // TODO: Look into memoiszing results so we don't hit the API again
```

Comment:
I think this might be super effective with pagination, as people might be pogo-sticking between pages, we don't even need to hit the API.